File: java/metodos/funcion_para_actualizar_contacto.java
package metodos;
import java.util.Scanner;
public class funcion_para_actualizar_contacto {
    public static boolean actualizarContacto(String[] contactos, String nombreAntiguo, String nombreNuevo) {
        for (int i = 0; i < contactos.length; i++) {
            if (contactos[i].equalsIgnoreCase(nombreAntiguo)) {
                contactos[i] = nombreNuevo;
                return true;
            }
        }
        return false;
    }
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        String[] contactos = {"Juan", "Maria", "Carlos", "Ana"};
        System.out.print("Ingrese el nombre del contacto a actualizar: ");
        String viejo = scanner.nextLine();
        System.out.print("Ingrese el nuevo nombre: ");
        String nuevo = scanner.nextLine();
        boolean actualizado = actualizarContacto(contactos, viejo, nuevo);
        if (actualizado) {
            System.out.println("Contacto actualizado correctamente.");
        } else {
            System.out.println("Contacto no encontrado.");
        }
        System.out.println("\n=== AGENDA ACTUALIZADA ===");
        for (int i = 0; i < contactos.length; i++) {
            System.out.println((i + 1) + ". " + contactos[i]);
        }
        scanner.close();
    }
}

File: visual_basic/metodos/funcion_para_actualizar_contacto/funcion_para_actualizar_contacto.vb
Imports System
Module funcion_para_actualizar_contacto
    Sub Main(args As String())
        Dim contactos As New List(Of String) From {
            "Juan", "Maria", "Carlos", "Ana"
        }
        ActualizarContacto(contactos)
        Console.WriteLine(vbCrLf & "=== LISTA ACTUALIZADA ===")
        For Each contacto As String In contactos
            Console.WriteLine(contacto)
        Next
        Console.ReadKey()
    End Sub
    Sub ActualizarContacto(ByRef lista As List(Of String))
        Console.WriteLine("Ingrese el nombre del contacto a actualizar:")
        Dim nombre As String = Console.ReadLine()
        Dim indice As Integer = -1
        For i As Integer = 0 To lista.Count - 1
            If lista(i).ToLower() = nombre.ToLower() Then
                indice = i
                Exit For
            End If
        Next
        If indice <> -1 Then
            Console.WriteLine("Ingrese el nuevo nombre:")
            Dim nuevoNombre As String = Console.ReadLine()
            lista(indice) = nuevoNombre
            Console.WriteLine("✅ Contacto actualizado")
        Else
            Console.WriteLine("❌ Contacto no encontrado")
        End If
    End Sub
End Module